fluentd/utils/sanitize_msg_chunks

```diff
@@ -0,0 +1,178 @@
+#!/bin/env ruby
+
+# Sanitize the buffer message chunks by unpacking.Verify the file buffers to confirm they
+# are deserializable and include the necessary
+# keys/values to push to Elasticsearch
+#
+require 'logger'
+require 'msgpack'
+require 'optparse'
+require 'ostruct'
+require 'fileutils'
+
+DEFAULT_BUFFER_DIR='/var/lib/fluentd'
+
+options = OpenStruct.new(
+  dir: '/var/lib/fluentd',
+  log_level: Logger::INFO,
+  dlq_dir: '/var/lib/fluentd/dlq',
+  index_field: 'viaq_index_name',
+  require: nil
+)
+
+OptionParser.new do |opts|
+  opts.banner = %Q(
+    Usage: #{File.basename(__FILE__)} [options]
+
+    Sanitize the buffer chunks to remove messages that
+    would otherwise block fluentd because the chunk is
+    corrupt and can not be properly unpacked.
+
+    Note: This operation is destructive and will replace
+    the existing buffer chunks.
+  )
+
+
+  opts.on("-d", "--directory DIRECTORY", "The directory where to find buffers to sanitize", "Default: #{options.dir}") do |dir|
+    options.dir = dir
+  end
+
+  opts.on("-i", "--index-field INDEX_FIELD", "The name of the key in the record that contains the index name",
+          "Default: #{options.index_field}") do | i |
+    options.index_field = i
+  end
+
+
+  opts.on("-q", "--dead-letter-queue DLQ", "The directory where to write unprocessable messages.",
+         "'DROP' - drop messages from buffer chunk", "Default: #{options.dlq_dir}") do |dir|
+    options.dlq_dir = dir
+  end
+
+  opts.on("-r", "--require EXTRA_HANDLER", "The full path to additional handler to further evaluate",
+          "each record. The file should contain a single function with the following name and args:",
+          "  check_record(logger, time, record)") do | r |
+    options.require = r
+  end
+
+  opts.on_tail("-v", "--verbose", "Run verbosely") do |v|
+    options.log_level = Logger::DEBUG
+  end
+end.parse!
+
+$logger = Logger.new(STDOUT)
+$logger.level = options.log_level
+$logger.formatter = proc do |s, date, prog, msg|
+  "[#{s}] - #{msg}\n"
+end
+
+$logger.debug("options: #{options}")
+
+def validate_args(options)
+  unless Dir.exists?(options.dir)
+      $logger.error("Unable to continue. Buffer directory does not exist: #{options.dir}")
+      return false
+  end
+  if options.index_field.strip.empty?
+      $logger.error("The index_field can not be empty.")
+      return false
+  end
+  if options.dlq_dir != 'DROP'
+     unless options.dlq_dir.start_with?('/')
+        $logger.error("The dead letter queue must be either 'DROP' or an absolute path: #{options.dlq_dir}")
+        return false
+     end
```

Comment:
check for `Dir.exists?(options.dlq_dir)` here too?

Reply:
We create it if it doesnt exist